Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -0,0 +1,77 @@
+/****************************************************************************
+* Copyright 2020, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import Foundation
+
+/// A struct for user contexts that the SDK will use to make decisions for.
+public struct OptimizelyUserContext {
+    weak var optimizely: OptimizelyClient?
+    var userId: String
+    
+    var atomicAttributes: AtomicProperty<[String: Any]>
```

Comment:
I love that we are using AtomicProperty.  Do you think there is a chance of a lot of these being created?  The reason I ask is that the more atomic properties we create, the more potential thread resources are being used (i.e. each concurrent dispatch queue created takes up resources).  So, I would suggest that we change the AtomicProperty to use a global concurrent queue (which will scale threads as needed) to save resources.  I'm going to create a separate PR for that right now.

Reply:
It's not expected that we have many user-context (one in most cases), so no need here. But it can be helpful for other cases.